src/training.py

```diff
@@ -115,58 +115,35 @@ def evaluate(
 
 def train_model(
         x_dim: int,
-        seq_len: int,
         train_loader: DataLoader,
         valid_loader: DataLoader,
         test_loader: DataLoader,
         # TODO only pass the model, not all the parameters
         # Begin model specific parameters
         d_model: int = 128,
-        num_heads: int = 2,
-        num_encoders: int = 8,
-        num_outputs: int = 3,
-        num_lstm_layers: int = 3,
-        lstm_dim: int = 256,
-        k: int = 64,
-        fc_dim: int = 512,
-        fc_dropout: float = 0.2,
-        lr: float = 0.0002,
         time_idx: list[int] = None,
         # End model specific parameters
         # TODO These are top level training parameters, should be moved to a config
-        optimizer: str = "adam",
-        scheduler: str = "plateau",
-        criterion: str = "ce",
-        epochs: int = 20,
         train_label_ct: th.Tensor = None,
-        model_class: AbstractModel = STTransformer,
+        model_class: AbstractModel = None,
         writer: SummaryWriter = None,
-        **kwargs
+        model_params = None,
+        trainer_params = None
 ):
     """Train a model and test the methods"""
     device = th.device('cuda' if th.cuda.is_available() else 'cpu')
 
-    model = STTransformer(
+    model = model_class(
         d_features=x_dim,
```

Comment:
do we want to get these as base level (in the abstract class) params? anything, like device, d_features, possibly batch size, input/output sizes that every model is going to need could be set up there... (I can make that change once this is in place, I'm mostly thinking out loud :D )

Reply:
Yeah that would make sense. Anything that has to be the same across models can be in the base class. I want to keep things tunable per model however so that when we run the "final test", params can be optimized to give the best results for each method.